File: 104-Maximum-Depth-of-Binary-Tree.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        else:
            lDepth = self.maxDepth(root.left)
            rDepth = self.maxDepth(root.right)
            return max(lDepth, rDepth) + 1

File: 104-Maximum-Depth-of-Binary-Tree.cpp
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if (!root)
            return 0;

        int lMaxDepth = maxDepth(root->left);
        int rMaxDepth = maxDepth(root->right);

        return max(lMaxDepth, rMaxDepth) + 1;
    }
};